### PR TITLE
Get updates for github issues based on last_modified (#259)

### PR DESCRIPTION
**Pull request by ksy36**
This is a follow-up for https://github.com/mozilla/bugbug/pull/2242 (fetching and saving updates if `last_modified` is available and if not, download all issues)

### CODE REVIEW

File: bugbug/github.py [hunk 1]
```diff
@@ -32,6 +32,10 @@ def get_issues() -> Iterator[IssueDict]:
     yield from db.read(GITHUB_ISSUES_DB)
 
 
+def delete_issues(match):
```
Comment (marco-c) [inline bugbug/github.py:35]: Can you add type annotations here? It should be something like `def delete_issues(match: Callable[IssueDict, bool]) -> None`.

(you'd need to define IssueDict like https://github.com/mozilla/bugbug/blob/e9106d8b2e1889b49fd0177d8f4537c83383d853/bugbug/bugzilla.py#L19).

File: bugbug/github.py [hunk 2]
```diff
@@ -79,9 +83,34 @@ def get_start_page() -> int:
     return int(count / PER_PAGE) + 1
 
 
+def fetch_issues_updated_since_timestamp(
+    owner: str, repo: str, state: str, since: str, retrieve_events: bool = False
+):
```
Comment (marco-c) [inline bugbug/github.py:88]: This is missing the return type annotation

File: tests/test_github.py [hunk 3]
```diff
@@ -82,3 +82,58 @@ def test_download_issues():
     )
 
     github.download_issues("webcompat", "web-bugs", "all")
+
+
+def test_download_issues_updated_since_timestamp():
```
Comment (marco-c) [inline tests/test_github.py:87]: Nit: `def test_download_issues_updated_since_timestamp() -> None:` so the function gets type checked.

**Review by marco-c — COMMENTED** [review]
Looks great overall!

File: bugbug/github.py (hunk 1 above)
Comment (ksy36) [reply bugbug/github.py]: Oh right, thanks 👍

**Comment by ksy36** [comment]
Thanks! I've pushed the fixes, could you please take another look @marco-c ?

File: tests/test_github.py [hunk 4]
```diff
@@ -46,15 +48,15 @@ def test_fetch_issues_with_events():
     # Mock events request
     responses.add(responses.GET, TEST_EVENTS_URL, json=expected_events, status=200)
 
-    # Assert that response with events has expected format
-    response_with_events = github.fetch_issues(TEST_URL, True)
     expected_with_events = expected
     expected_with_events[0]["events"] = expected_events
 
+    # Assert that response with events has expected format
+    response_with_events = github.fetch_issues(TEST_URL, True)
```
Comment (marco-c) [inline tests/test_github.py:55]: Nit: why did you move this?

File: bugbug/github.py [hunk 5]
```diff
@@ -79,9 +83,34 @@ def get_start_page() -> int:
     return int(count / PER_PAGE) + 1
 
 
+def fetch_issues_updated_since_timestamp(
+    owner: str, repo: str, state: str, since: str, retrieve_events: bool = False
+) -> list:
```
Comment (marco-c) [inline bugbug/github.py:88]: This can be `list[IssueDict]`

File: bugbug/github.py [hunk 6]
```diff
@@ -55,7 +59,7 @@ def fetch_events(events_url: str) -> str:
 
 def fetch_issues(
     url: str, retrieve_events: bool, params: dict = None
-) -> Tuple[str, dict]:
+) -> Tuple[list, dict]:
```
Comment (marco-c) [inline bugbug/github.py:62]: This can be list[IssueDict]

**Review by marco-c — DISMISSED** [review]
Thanks!

File: tests/test_github.py (hunk 4 above)
Comment (ksy36) [reply tests/test_github.py:55]: I've separated building expected data from the actual response to make it a bit more readable

Comment (marco-c) [reply tests/test_github.py:55]: Ok, WFM.

File: bugbug/github.py (hunk 5 above)
Comment (ksy36) [reply bugbug/github.py:88]: Getting an error with using `list`, `Class 'type' doesn't define '__getItem__', so the '[]' operator cannot be used on its instances`, so going to use `typing.List` instead of `list`

**Review by marco-c — APPROVED** [review]

**Comment by marco-c** [comment]
The test failures are not this PR's fault, but a problem introduced with 0094619b67c74a8d5d188918a009b21e47c9c0b2.

**Comment by ksy36** [comment]
Thanks!
